src/huggingface_hub/hf_api.py

```diff
@@ -743,7 +743,8 @@ def create_repo(
 
         json = {"name": name, "organization": organization, "private": private}
         if repo_type is not None:
-            json["type"] = repo_type
+            if repo_type != "model":
+                json["type"] = repo_type
```

Comment:
For clarity I'd keep the same logic you implemented everywhere else, i.e., keeping the `if repo_type ... None ... "model"`:

```diff
-        if repo_type is not None:
+        if repo_type is not None and repo_type != "model":
-            if repo_type != "model":
-                json["type"] = repo_type
+            json["type"] = repo_type
```

Reply:
Adjusted